src/Views/OperatingSystemView.vala

```diff
@@ -248,6 +257,33 @@ public class About.OperatingSystemView : Gtk.Grid {
         }
     }
 
+    private bool confirm_reboot_to_firmware_setup_action () {
+        var dialog = new Granite.MessageDialog.with_image_from_icon_name (
+            _("Are you sure you want to Reboot?"),
+            _("This will close all open applications and turn off this device."),
+            "system-shutdown",
+            Gtk.ButtonsType.CANCEL
+        );
+        dialog.transient_for = (Gtk.Window) get_toplevel ();
+
+        var continue_button = dialog.add_button (_("Reboot"), Gtk.ResponseType.ACCEPT);
+        continue_button.get_style_context ().add_class (Gtk.STYLE_CLASS_DESTRUCTIVE_ACTION);
+
+        var result = dialog.run ();
+        dialog.destroy ();
+
+        return result == Gtk.ResponseType.ACCEPT;
+    }
+
+    private void reboot_to_firmware_setup_clicked () {
+        if (confirm_reboot_to_firmware_setup_action ()) {
```

Comment:
Kind of going with the other thing, we should connect to `dialog.response ()` since we can't get a return value from `run ()` anymore. This is kind of weird to have as a single-use function anyways